Tasks/AzureResourceGroupDeployment/models/DeployAzureRG.ts

```diff
@@ -4,6 +4,32 @@
 import tl = require("vsts-task-lib/task");
 import msRestAzure = require("./../operations/azure-rest/azure-arm-common");
 
+class TokenCredentials {
+    private hostUrl: string;
+    private patToken: string;
+
+    constructor(hostUrl: string, patToken: string){
+        if (!Boolean(hostUrl) || typeof hostUrl.valueOf() !== 'string') {
```

Comment:
these two checks must be done before calling this constructor as well. And error message should clearly say that endpoint's url/token field must not be empty.